lib/container_orchestrator.rb

```diff
@@ -95,6 +95,14 @@ def my_pod
     get_pod_by_namespace_and_hostname(my_namespace, ENV["HOSTNAME"])
   end
 
+  def my_node_affinity_arch_values
+    ContainerOrchestrator.new.my_pod.spec.affinity&.nodeAffinity&.requiredDuringSchedulingIgnoredDuringExecution&.nodeSelectorTerms&.each do |i|
+      i.matchExpressions&.each { |a| return(a.values) if a.key == "kubernetes.io/arch" }
+    end
+
+    return ["amd64"]
```

Comment:
I added the same default value in the operator.  I'm not sure if it's better to have a default or not set the affinity at all.